presentation-compiler/src/main/scala/org/alephium/ralph/lsp/pc/sourcecode/SourceLocation.scala

```diff
@@ -44,10 +44,15 @@ object SourceLocation {
    */
   sealed trait GoToDef extends GoTo
 
+  /**
+   * Result types for renaming location search results.
+   */
+  sealed trait Rename extends GoTo
+
   /**
    * Result types for GoTo references location search results.
    */
-  sealed trait GoToRef extends GoTo
+  sealed trait GoToRef extends Rename
```

Comment:
it sounds weird that `GoToRef`extends `Rename`, is it expected?

Reply:
Good catch. The only reason for this is to save a transformation because `rename` returns the [same result type](https://github.com/alephium/ralph-lsp/pull/314/files#diff-723a4cc88a96e60434eef6c97ad35b90b026a84709f80bd5feda76b0c2830b4dR61) as `GoToRef`. Having `GoToRef` as a subtype of `Rename` means we can return the `GoToRef` results from `rename` immediately, without the need to transform them to the `GoTo.Rename` type. 

If in the future the `Rename` result types needs additional types, then the transformation will be applied and we can seperate them:

```scala
sealed trait Rename  extends GoTo
sealed trait GoToRef extends GoTo
```

If you prefer, I can seperate them now? I was just saving it from doing a transformation, which really isn't that expensive.

Reply:
ok let's keep it like this for now then.